hashicorp/terraform
# Remote Backend: Support latest cost-estimate API

Updates the remote backend to use the latest cost-estimate API, which works slightly differently. There are some unrelated changes to the mocks due to the go-tfe version bump.

To test/run this:
- setup your config using the remote backend
- add an AWS instance in the config
- `XC_OS="darwin" XC_ARCH="amd64" make bin` (for darwin machines) in this branch in terraform
- `<your terraform dir>/pkg/darwin_amd64/terraform plan` in the root of your config
- output should show plan results, then cost estimation results

--------

```HCL
terraform {
  backend "remote" {
    hostname = "app.terraform.io"
    organization = "pthrasher_v2"
    token = "<your TFC user token>"

    workspaces {
      name = "tfe-remote-backend-example"
    }
  }
}

provider "aws" {
  region  = "us-east-1"
}

resource "aws_instance" "basic-A" {
  ami           = "ami-62131702"
  instance_type = "t3.nano"
}
```
